clang/lib/Format/QualifierAlignmentFixer.cpp

```diff
@@ -412,6 +414,15 @@ const FormatToken *LeftRightQualifierAlignmentFixer::analyzeLeft(
   // The case `const long long volatile int` -> `const volatile long long int`
   // The case `long volatile long int const` -> `const volatile long long int`
   if (TypeToken->isTypeName(LangOpts)) {
+    for (const auto *Prev = TypeToken->getPreviousNonComment();
+         Prev && Prev->is(tok::coloncolon);
+         Prev = Prev->getPreviousNonComment()) {
+      TypeToken = Prev;
+      Prev = Prev->getPreviousNonComment();
```

Comment:
I will replace `getPreviousNonComment()` with `Previous` for simplicity and better performance as we really don't need to be concerned about code like `::/**/std/**/::/**/int64_t`.